### PR TITLE
Restrict repair rates of  Orbital Drydock

### PR DESCRIPTION
**Pull request by dbenage-cx**
- [x] Finalized formula for repair rate and handicaps
- [x] Update pedia entries and sitreps

Adjust Orbital Drydock per Issue https://github.com/freeorion/freeorion/issues/539
- remove Infrastructure penalty
- add a 1 turn wait period until repairs
- repairs require minimum happiness of 5
- increased happiness increases repair rate
- repair rate is a percentage of current ship structure (minimum +5)
- separate sitrep/icon for completed repairs

### CODE REVIEW

**Comment by MatGB** [comment]
Initial testing makes me very keen on this as a first pass, but before I merge it in can someone on the @freeorion/ai-team give an opinion on whether the AI should be able to handle it? It doesn't seem to be making a difference from observation but I know repair is a problem for the AI sometimes.

File: default/stringtables/en.txt [hunk 1]
```diff
@@ -4866,8 +4866,10 @@ At %planet%: depopulation is complete.
...
 Ethnic Cleansing
 
-EFFECT_DRYDOCK_SHIP_REPAIR
-On %planet%: the %building% repaired the %ship%.
+EFFECT_DRYDOCK_SHIP_REPAIR_PARTIAL
+On %planet%: partial repairs were performed on the %ship% by the %building%.
+EFFECT_DRYDOCK_SHIP_REPAIR_COMPLETE
+On %planet%: the %building% completed repairs on the %ship%.
 EFFECT_DRYDOCK_SHIP_REPAIR_LABEL
 Drydock Repair
```
Comment (MatGB) [inline default/stringtables/en.txt:4918]: The repair label needs updating for both, as far as I know it needs separate labels else you get two identical options to select/unselect.

File: default/stringtables/en.txt [hunk 2]
```diff
@@ -4866,8 +4866,10 @@ At %planet%: depopulation is complete.
 EFFECT_CONC_CAMP_COMLETE_LABEL
 Ethnic Cleansing
 
-EFFECT_DRYDOCK_SHIP_REPAIR
-On %planet%: the %building% repaired the %ship%.
+EFFECT_DRYDOCK_SHIP_REPAIR_PARTIAL
+On %planet%: partial repairs were performed on the %ship% by the %building%.
```
Comment (MatGB) [inline default/stringtables/en.txt]: Needs to be worded the same way as complete, ie "the %building% partially repaired the %ship%." That's to ensure that fleet/shipnames are always clickable at the end of the line for consistency

**Comment by Morlic-fo** [comment]
Currently AI won't be able to handle it, but (preliminary) fix shouldn't be too hard... I'll post PR soon.

**Comment by Morlic-fo** [comment]
You should probably wait with merging this until @Cjkjvfnby agrees with #554.

**Comment by Morlic-fo** [comment]
AI adapatations are ready. If you merge this, merge #554 as well.

**Comment by Vezzra** [comment]
@MatGB, can you hold off for a day? I'd want to take a look at this tomorrow (Tuesday), didn't have a chance to do it before.

**Comment by MatGB** [comment]
Wasn't expecting the AI adjustments that quickly anyway—it plays OK, but I'm not sure the repair rate is high enough.

On the other hand once the rest of the damage control techs are toned down the drydocks will feel faster anyway.

**Comment by geoffthemedio** [comment]
Wasn't there extensive discussion about repair rates on the forums, with comments that it's preferable to always have instant full repair to avoid needing to wait and keep track of which ships are repaired or not? Was it decided that the strategic problems of instant repair outweigh this concern?

**Comment by MatGB** [comment]
There were discussions that involved some people stating that view but that was never the final decision, specifically instant full repair was re-introduced (by me at Vezzra's suggestion) from the previous 10-per-turn system as a very short term measure that would work for the then imminent Release which would then be reworked during the next (this) cycle.

Worth noting the main thrust of this change is to stop drydocks working immediately on conquest and to start getting better, to massively reduce the steamroller effect they current allow. Drydocks in 'safe' home territory will work at full strength and give full repair on all but the most heavily damaged ships.

Instant repair was only implemented as an agreed stop gap solution, and this is part of a wider (agreed) pass to redo the damage control techs and introduce damage control ship parts.

**Comment by Vezzra** [comment]
Yes, there was this discussion, and still is, as [I've just posted](http://www.freeorion.org/forum/viewtopic.php?p=83248#p83248) on the [Orbital Drydock revision thread](http://www.freeorion.org/forum/viewtopic.php?f=15&t=10007) on forum, once again restating all my concerns I have with that idea. As @MatGB noted, there hasn't been a final decision on that yet, going back to full repair was just a temporary measure because the 10 turn delay for drydock repairs which we had then didn't work at all.

> Worth noting the main thrust of this change is to stop drydocks working immediately on conquest and to start getting better, to massively reduce the steamroller effect they current allow.

Just requiring a min happiness of 5 (and optionally introducing that one turn delay after arrvial at a drydock, if you think that really necessary) will already achieve that. This is yet another case where I'd just go with the very simple solution first and see where it gets us, before resorting to more complicated (potentially issue-prone) solutions.

> Drydocks in 'safe' home territory will work at full strength and give full repair on all but the most heavily damaged ships.

My first quick test showed that it does not work like you describe here. I sent two early game ships against a Maintenance Ship and let them sustain some damage, returned them to my capitol (the only drydock in early game), where it took 2 turns for the first and 3 for the second ship to get fully repaired. This second 25 happiness treshold for the full repair rate doesn't work out IMO (see [my reply](http://www.freeorion.org/forum/viewtopic.php?p=83248#p83248) on the forum).

**Comment by Vezzra** [comment]
The little bit of quick testing I've been able to do indicates that this seems to work. The formula as currently implemented meets the consensus we've reached so far. Anything that should be adjusted further?

If not, I propose to merge this, so this will receive more thorough testing. If no one objects I'm going to merge before I produce the upcoming test builds.

**Comment by dbenage-cx** [comment]
Restructured commits, no content changes.

**Comment by MatGB** [comment]
I was planning on testing it this afternoon but got distracted by Geoff's Trith change, but yeah, if you're happy put it in and we can tweak if after feedback, assuming the @freeorion/ai-team have no objections.

**Pull request merged**